components/brave_wallet_ui/page/screens/swap/swap.tsx

```diff
@@ -93,13 +98,21 @@ export const Swap = () => {
 
   const [getTokenBalancesForChainId] = useLazyGetTokenBalancesForChainIdQuery()
   const getTokenBalancesForChainIdWrapped = React.useCallback(
-    async (contracts: string[], address: string, coin: BraveWallet.CoinType, chainId: string) => {
-      return await getTokenBalancesForChainId({
-        contracts,
-        address,
-        coin,
-        chainId
-      }).unwrap()
+    async (contracts: string[], address: string, coin: CoinType, chainId: string) => {
+      return await getTokenBalancesForChainId(
+        coin === CoinTypes.ETH
+          ? {
+              contracts,
+              address,
+              coin,
+              chainId
+            }
+          : {
+              address,
+              coin,
+              chainId
```

Comment:
Prefer throwing an exception for anything other than ETH, and add support in a future PR. In any case, it'll need changes after the Swap migration PR is merged.

Reply:
now throwing an exception for anything other than ETH